vnext/Microsoft.ReactNative.Cxx/JSValueReader-WinRT.h

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <CppWinRTIncludes.h>
+#include <UI.Xaml.Markup.h>
+#include <chrono>
+
+#include "JSValue.h"
+#include "JSValueTreeReader.h"
+#include "StructInfo.h"
+
+#include "winrt/Microsoft.ReactNative.h"
+
+#include <string>
+
+namespace winrt::Microsoft::ReactNative {
+inline BYTE GetAFromArgb(DWORD v) {
+  return ((BYTE)((v & 0xFF000000) >> 24));
+}
+inline BYTE GetRFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x00FF0000) >> 16));
+}
+inline BYTE GetGFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x0000FF00) >> 8));
+}
+inline BYTE GetBFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x000000FF)));
+}
+
+inline ui::Color ColorFromArgb(const JSValue &value) {
+  auto argb = static_cast<DWORD>(value.AsInt64());
+  return winrt::ColorHelper::FromArgb(GetAFromArgb(argb), GetRFromArgb(argb), GetGFromArgb(argb), GetBFromArgb(argb));
+}
+
+inline ui::Color UndefinedColor() noexcept {
+  static auto c = ColorHelper::FromArgb(1, 0, 0, 0);
+  return c;
+}
+
+inline xaml::Media::SolidColorBrush UndefinedSolidColorBrush() noexcept {
+  static auto b = xaml::Media::SolidColorBrush(UndefinedColor());
+  return b;
+}
+
+inline void ReadValue(IJSValueReader const &reader, /*out*/ ui::Color &value) {
+  value = UndefinedColor();
+  if (reader.ValueType() == JSValueType::Int64) {
+    auto argb = static_cast<DWORD>(reader.GetInt64());
+    value =
+        winrt::ColorHelper::FromArgb(GetAFromArgb(argb), GetRFromArgb(argb), GetGFromArgb(argb), GetBFromArgb(argb));
+  }
+}
+
+struct ColorComp {
```

Comment:
>Comp [](start = 12, length = 4)

Consider using the full name Comparer or something else along these lines per Microsoft naming guidelines. [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions)

Reply:
From the usage I understood that this really an implementation of the std::less algorithm. A much better solution would be to define the operator<, and then use std::less<ui::Color> in the std::map which will pickup the operator< implementation. I guess we can make this operator constexpr:
```cpp
constexpr bool operator<(ui::Color const &lhs, ui::Color const &rhs) {...}
```